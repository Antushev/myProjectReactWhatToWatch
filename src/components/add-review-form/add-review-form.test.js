import React from 'react';
import renderer from 'react-test-renderer';

import AddReviewForm from './add-review-form.jsx';

describe(`AddReViewFormComponent`, () => {
  it(`AddReviewFormComponentSnapshot with isButtonBlocked true`, () => {
    const tree = renderer
      .create(
          <AddReviewForm
            rating={4}
            isLoadingComment={false}
            isButtonBlocked={true}
            isErrorLoadingComment={false}
            onRatingChange={() => {}}
            onTextChange={() => {}}
            onSubmitClick={() => {}}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`AddReviewFormComponentSnapshot with isButtonBlocked false`, () => {
    const tree = renderer
      .create(
          <AddReviewForm
            rating={4}
            isLoadingComment={false}
            isButtonBlocked={false}
            isErrorLoadingComment={false}
            onRatingChange={() => {}}
            onTextChange={() => {}}
            onSubmitClick={() => {}}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`AddReviewFormComponentSnapshot with isLoadingComment true`, () => {
    const tree = renderer
      .create(
          <AddReviewForm
            rating={4}
            isLoadingComment={true}
            isButtonBlocked={false}
            isErrorLoadingComment={true}
            onRatingChange={() => {}}
            onTextChange={() => {}}
            onSubmitClick={() => {}}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`AddReviewFormComponentSnapshot with isLoadingComment true and isButtonBlocked true`, () => {
    const tree = renderer
      .create(
          <AddReviewForm
            rating={4}
            isLoadingComment={true}
            isButtonBlocked={true}
            isErrorLoadingComment={false}
            onRatingChange={() => {}}
            onTextChange={() => {}}
            onSubmitClick={() => {}}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
