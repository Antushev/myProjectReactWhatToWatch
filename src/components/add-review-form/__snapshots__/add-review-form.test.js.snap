// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddReViewFormComponent AddReviewFormComponentSnapshot with isButtonBlocked false 1`] = `
<form
  action="#"
  className="add-review__form"
  onSubmit={[Function]}
>
  <div
    className="rating"
  >
    <div
      className="rating__stars"
    >
      <input
        className="rating__input visually-hidden"
        defaultChecked={true}
        disabled={false}
        hidden={true}
        id="star-0"
        name="rating"
        onChange={[Function]}
        type="radio"
        value={0}
      />
      <label
        className="rating__label visually-hidden"
        htmlFor="star-0"
      >
        Rating 0
      </label>
      <input
        className="rating__input"
        defaultChecked={false}
        disabled={false}
        id="star-1"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="1"
      />
      <label
        className="rating__label"
        htmlFor="star-1"
      >
        Rating 
        1
      </label>
      <input
        className="rating__input"
        defaultChecked={false}
        disabled={false}
        id="star-2"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="2"
      />
      <label
        className="rating__label"
        htmlFor="star-2"
      >
        Rating 
        2
      </label>
      <input
        className="rating__input"
        defaultChecked={false}
        disabled={false}
        id="star-3"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="3"
      />
      <label
        className="rating__label"
        htmlFor="star-3"
      >
        Rating 
        3
      </label>
      <input
        className="rating__input"
        defaultChecked={false}
        disabled={false}
        id="star-4"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="4"
      />
      <label
        className="rating__label"
        htmlFor="star-4"
      >
        Rating 
        4
      </label>
      <input
        className="rating__input"
        defaultChecked={false}
        disabled={false}
        id="star-5"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="5"
      />
      <label
        className="rating__label"
        htmlFor="star-5"
      >
        Rating 
        5
      </label>
    </div>
  </div>
  <div
    className="add-review__text"
  >
    <textarea
      className="add-review__textarea"
      disabled={false}
      id="review-text"
      maxLength="400"
      minLength="50"
      name="review-text"
      onChange={[Function]}
      placeholder="Review text"
    />
    <div
      className="add-review__submit"
    >
      <button
        className="add-review__btn"
        color="#F00"
        disabled={false}
        style={
          Object {
            "color": "#866866",
            "cursor": "pointer",
          }
        }
        type="submit"
      >
        Post
      </button>
    </div>
  </div>
  
</form>
`;

exports[`AddReViewFormComponent AddReviewFormComponentSnapshot with isButtonBlocked true 1`] = `
<form
  action="#"
  className="add-review__form"
  onSubmit={[Function]}
>
  <div
    className="rating"
  >
    <div
      className="rating__stars"
    >
      <input
        className="rating__input visually-hidden"
        defaultChecked={true}
        disabled={false}
        hidden={true}
        id="star-0"
        name="rating"
        onChange={[Function]}
        type="radio"
        value={0}
      />
      <label
        className="rating__label visually-hidden"
        htmlFor="star-0"
      >
        Rating 0
      </label>
      <input
        className="rating__input"
        defaultChecked={false}
        disabled={false}
        id="star-1"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="1"
      />
      <label
        className="rating__label"
        htmlFor="star-1"
      >
        Rating 
        1
      </label>
      <input
        className="rating__input"
        defaultChecked={false}
        disabled={false}
        id="star-2"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="2"
      />
      <label
        className="rating__label"
        htmlFor="star-2"
      >
        Rating 
        2
      </label>
      <input
        className="rating__input"
        defaultChecked={false}
        disabled={false}
        id="star-3"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="3"
      />
      <label
        className="rating__label"
        htmlFor="star-3"
      >
        Rating 
        3
      </label>
      <input
        className="rating__input"
        defaultChecked={false}
        disabled={false}
        id="star-4"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="4"
      />
      <label
        className="rating__label"
        htmlFor="star-4"
      >
        Rating 
        4
      </label>
      <input
        className="rating__input"
        defaultChecked={false}
        disabled={false}
        id="star-5"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="5"
      />
      <label
        className="rating__label"
        htmlFor="star-5"
      >
        Rating 
        5
      </label>
    </div>
  </div>
  <div
    className="add-review__text"
  >
    <textarea
      className="add-review__textarea"
      disabled={false}
      id="review-text"
      maxLength="400"
      minLength="50"
      name="review-text"
      onChange={[Function]}
      placeholder="Review text"
    />
    <div
      className="add-review__submit"
    >
      <button
        className="add-review__btn"
        color="#F00"
        disabled={true}
        style={
          Object {
            "color": "#e1b0b2",
            "cursor": "default",
          }
        }
        type="submit"
      >
        Post
      </button>
    </div>
  </div>
  
</form>
`;

exports[`AddReViewFormComponent AddReviewFormComponentSnapshot with isLoadingComment true 1`] = `
<form
  action="#"
  className="add-review__form"
  onSubmit={[Function]}
>
  <div
    className="rating"
  >
    <div
      className="rating__stars"
    >
      <input
        className="rating__input visually-hidden"
        defaultChecked={true}
        disabled={true}
        hidden={true}
        id="star-0"
        name="rating"
        onChange={[Function]}
        type="radio"
        value={0}
      />
      <label
        className="rating__label visually-hidden"
        htmlFor="star-0"
      >
        Rating 0
      </label>
      <input
        className="rating__input"
        defaultChecked={false}
        disabled={true}
        id="star-1"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="1"
      />
      <label
        className="rating__label"
        htmlFor="star-1"
      >
        Rating 
        1
      </label>
      <input
        className="rating__input"
        defaultChecked={false}
        disabled={true}
        id="star-2"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="2"
      />
      <label
        className="rating__label"
        htmlFor="star-2"
      >
        Rating 
        2
      </label>
      <input
        className="rating__input"
        defaultChecked={false}
        disabled={true}
        id="star-3"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="3"
      />
      <label
        className="rating__label"
        htmlFor="star-3"
      >
        Rating 
        3
      </label>
      <input
        className="rating__input"
        defaultChecked={false}
        disabled={true}
        id="star-4"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="4"
      />
      <label
        className="rating__label"
        htmlFor="star-4"
      >
        Rating 
        4
      </label>
      <input
        className="rating__input"
        defaultChecked={false}
        disabled={true}
        id="star-5"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="5"
      />
      <label
        className="rating__label"
        htmlFor="star-5"
      >
        Rating 
        5
      </label>
    </div>
  </div>
  <div
    className="add-review__text"
  >
    <textarea
      className="add-review__textarea"
      disabled={true}
      id="review-text"
      maxLength="400"
      minLength="50"
      name="review-text"
      onChange={[Function]}
      placeholder="Review text"
    />
    <div
      className="add-review__submit"
    >
      <button
        className="add-review__btn"
        color="#F00"
        disabled={false}
        style={
          Object {
            "color": "#866866",
            "cursor": "pointer",
          }
        }
        type="submit"
      >
        Post
      </button>
    </div>
  </div>
  <p>
    Комментарий загружается
  </p>
</form>
`;

exports[`AddReViewFormComponent AddReviewFormComponentSnapshot with isLoadingComment true and isButtonBlocked true 1`] = `
<form
  action="#"
  className="add-review__form"
  onSubmit={[Function]}
>
  <div
    className="rating"
  >
    <div
      className="rating__stars"
    >
      <input
        className="rating__input visually-hidden"
        defaultChecked={true}
        disabled={true}
        hidden={true}
        id="star-0"
        name="rating"
        onChange={[Function]}
        type="radio"
        value={0}
      />
      <label
        className="rating__label visually-hidden"
        htmlFor="star-0"
      >
        Rating 0
      </label>
      <input
        className="rating__input"
        defaultChecked={false}
        disabled={true}
        id="star-1"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="1"
      />
      <label
        className="rating__label"
        htmlFor="star-1"
      >
        Rating 
        1
      </label>
      <input
        className="rating__input"
        defaultChecked={false}
        disabled={true}
        id="star-2"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="2"
      />
      <label
        className="rating__label"
        htmlFor="star-2"
      >
        Rating 
        2
      </label>
      <input
        className="rating__input"
        defaultChecked={false}
        disabled={true}
        id="star-3"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="3"
      />
      <label
        className="rating__label"
        htmlFor="star-3"
      >
        Rating 
        3
      </label>
      <input
        className="rating__input"
        defaultChecked={false}
        disabled={true}
        id="star-4"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="4"
      />
      <label
        className="rating__label"
        htmlFor="star-4"
      >
        Rating 
        4
      </label>
      <input
        className="rating__input"
        defaultChecked={false}
        disabled={true}
        id="star-5"
        name="rating"
        onChange={[Function]}
        type="radio"
        value="5"
      />
      <label
        className="rating__label"
        htmlFor="star-5"
      >
        Rating 
        5
      </label>
    </div>
  </div>
  <div
    className="add-review__text"
  >
    <textarea
      className="add-review__textarea"
      disabled={true}
      id="review-text"
      maxLength="400"
      minLength="50"
      name="review-text"
      onChange={[Function]}
      placeholder="Review text"
    />
    <div
      className="add-review__submit"
    >
      <button
        className="add-review__btn"
        color="#F00"
        disabled={true}
        style={
          Object {
            "color": "#e1b0b2",
            "cursor": "default",
          }
        }
        type="submit"
      >
        Post
      </button>
    </div>
  </div>
  <p>
    Комментарий загружается
  </p>
</form>
`;
